docs/configuration.md

```diff
@@ -680,6 +681,95 @@ window.$docsify = {
 };
 ```
 
+## routes
+
+- Type: `Object`
+
+Define "virtual" routes that can provide content dynamically. A route is a map between the expected path, to either a string or a function. If the mapped value is a string, it is treated as markdown and parsed accordingly. If it is a function, it is expected to return markdown content.
+
+A route function receives up to three parameters:
+1. `route` - the path of the route that was requested (e.g. `/bar/baz`)
+2. `matched` - the [`RegExpMatchArray`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match) that was matched by the route (e.g. for `/bar/(.+)`, you get `['/bar/baz', 'baz']`)
+3. `next` - this is a callback that you may call when your route function is async
+
+Do note that order matters! Routes are matched the same order you declare them in, which means that in cases where you have overlapping routes, you might want to list the more specific ones first.
+
+```js
+window.$docsify = {
+  routes: {
+    // Basic match w/ return string
+    '/foo': '# Custom Markdown',
+
+    // RegEx match w/ synchronous function
+    '/bar/(.*)': function(route, matched) {
+      return '# Custom Markdown';
+    },
+
+    // RegEx match w/ asynchronous function
+    '/baz/(.*)': function(route, matched, next) {
+       try {
+        // Async task(s)...
+      } catch (err) {
+        // ...
+      } finally {
+        next('# Custom Markdown');
+      }
+    }
+  }
+}
+```
+
+Other than strings, route functions can return a falsy value (`null` \ `undefined`) to indicate that they ignore the current request:
+
+```js
+window.$docsify = {
+  routes: {
+    // accepts everything other than dogs (synchronous)
+    '/pets/(.+)': function(route, matched) {
+      if (matched[0] === 'dogs') {
+        return null;
+      } else {
+        return 'I like all pets but dogs';
+      }
+    }
+
+    // accepts everything other than cats (asynchronous)
+    '/pets/(.*)': function(route, matched, next) {
+      if (matched[0] === 'cats') {
+        next();
+      } else {
+        try {
```

Comment:
I personally would rather show new syntax that works reasonably in all current browsers, and let people who _need_ to support older browsers figure what old syntax to use. That's just my preference.

We're just making our (limited time maintainer) lives more difficult for a very small minority of users.

Reply:
I personally believe there's not much sense in not teaching people async/await these days. It has been _7 years_ since ES2015.

Reply:
@jhildenbiddle wdyt?

Reply:
@trusktr makes some valid points. Here's my $0.02...

> Note that async tasks that throw errors without using async/await syntax, will not be caught by the try-catch block.

Yep. Good catch (pun intended). I think I was a little hyper-focused on showing a simple example with try/catch block for async tasks to prevent uncaught errors from breaking sites and missed the fact that my simple example doesn't actually work. 🤦‍♂️ 

> I personally would rather show new syntax that works reasonably in all current browsers, and let people who need to support older browsers figure what old syntax to use. That's just my preference.

Let's accommodate both scenarios by showing an example that uses an ES6+ feature that easily be polyfilled by those who care about legacy browser support:

```js
// RegEx match w/ asynchronous function
'/baz/(.*)': function(route, matched, next) {
  // Requires `fetch` polyfill for legacy browsers (https://github.github.io/fetch/)
  fetch('/api/users?id=12345')
    .then(function(response) {
      return response.text()
    })
    .then(function(body) {
      next('# Custom Markdown');
    })
    .catch(function(ex) {
      console.log('parsing failed', ex)
    });
}
 ```

The alternative is to simplify the example as much as possible and just focus on the addition of the `next` function for async tasks:

```js
// RegEx match w/ asynchronous function
'/baz/(.*)': function(route, matched, next) {
  // Async task(s)...
  next('# Custom Markdown');
}
 ```

I'm fine either way. I think I prefer the first option for this scenario because fetching remote data is a real-world use case for virtual routes.

Reply:
NP, updated the docs :)

Reply:
@trusktr Sound good to you?